registry/providerregistryprotocol/client.go

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) The OpenTofu Authors
+// SPDX-License-Identifier: MPL-2.0
+
+package providerregistryprotocol
```

Comment:
This is certainly not the most important thing and I don't feel strongly about it, but this seems rather long compared to typical Go package naming conventions, given that the package name often appears with high frequency in calling code. :thinking: 

The compromise I thought about is `package providerregistry`: still a _little_ long, but the noun "protocol" doesn't seem super crucial here since _everything_ in this module is related to protocols.

(Similar comment for `package providermirrorprotocol` too.)